### PR TITLE
Input validation

### PR DESCRIPTION
**Pull request by loansindi**
Allows entry of lowercase ICAO code without failing, and only allow four-character codes.

Update font size selection to a spinnerox.

### CODE REVIEW

**Comment by jobybenoit** [comment]
Font spinner box - working and tested
Block print if string is empty - working and tested
Change to uppercase - Not working

File: mainwindow.h.autosave [hunk 1]
```diff
@@ -0,0 +1,48 @@
+#ifndef MAINWINDOW_H
```
Comment (hef) [inline mainwindow.h.autosave:1]: probably don't want to check this file in and want to add *.autosave to .gitignore

File: moc_mainwindow.cpp [hunk 2]
```diff
@@ -0,0 +1,153 @@
+/****************************************************************************
```
Comment (hef) [inline moc_mainwindow.cpp]: I wouldn't check in moc_* files, I suggest adding them to .gitignore as well

**Review by hef — CHANGES_REQUESTED** [review]
Add the following to .gitingore:

*.autosave
moc_*
weather-qt
ui_*
*.user
*.o
Makefile

File: moc_predefs.h [hunk 3]
```diff
@@ -0,0 +1,389 @@
+#define __SSP_STRONG__ 3
```
Comment (hef) [inline moc_predefs.h]: delete this file from repo

File: mainwindow.ui [hunk 4]
```diff
@@ -51,9 +51,18 @@
         </widget>
        </item>
        <item>
-        <widget class="QLineEdit" name="fontSize">
-         <property name="text">
-          <string>10</string>
+        <widget class="QDoubleSpinBox" name="fontSize">
```
Comment (hef) [inline mainwindow.ui:54]: you replaced the lineedit with a numeric spinbox, was this intentional?

Comment (hef) [reply mainwindow.ui:54]: nevermind, I get it now

File: .gitignore [hunk 5]
```diff
@@ -0,0 +1,5 @@
+.autosave
+moc_*
+weather-qt
+ui_*
+*.user
```
Comment (hef) [inline .gitignore]: I'd add these as well, I forgot to include them in my initial comment

*.o
Makefile

**Review by hef — COMMENTED** [review]
ignore more things

**Comment by jobybenoit** [comment]
Tested, working

**Comment by jobybenoit** [comment]
New commits tested, No print on 404, Status bar reflects last action

**Pull request merged**